scripts/fetch_test_data.py

```diff
@@ -42,30 +44,65 @@ def fetch_datasets(search_facets: dict[str, Any], remove_ensembles: bool) -> pd.
     path_dict = cat.to_path_dict(prefer_streaming=False, minimal_keys=False)
 
     merged_df = cat.df.merge(pd.Series(path_dict, name="files"), left_on="key", right_index=True)
+    if time_span:
+        merged_df["time_start"] = time_span[0]
+        merged_df["time_end"] = time_span[1]
 
     return merged_df
 
 
-def decimate_dataset(dataset: xr.Dataset) -> xr.Dataset:
+def deduplicate_datasets(datasets: pd.DataFrame) -> pd.DataFrame:
+    """
+    Deduplicate a dataset collection.
+
+    Uses the metadata from the first dataset in each group,
+    but expands the time range to the min/max timespan of the group.
+
+    Parameters
+    ----------
+    datasets
+        The dataset collection
+
+    Returns
+    -------
+    pd.DataFrame
+        The deduplicated dataset collection spanning the times requested
+    """
+
+    def _deduplicate_group(group: pd.DataFrame) -> pd.DataFrame:
+        first = group.iloc[0].copy()
+        first.time_start = group.time_start.min()
+        first.time_end = group.time_end.max()
+
+        return first
+
+    return datasets.groupby("key").apply(_deduplicate_group, include_groups=False).reset_index()
+
+
+def decimate_dataset(dataset: xr.Dataset, time_span: tuple[str, str] | None) -> xr.Dataset | None:
     """
     Downscale the dataset to a smaller size.
 
     Parameters
     ----------
     dataset
         The dataset to downscale
+    time_span
+        The time span to extract from a dataset
 
     Returns
     -------
     xr.Dataset
         The downscaled dataset
     """
-    spatial_downscale = dataset.interp(lat=dataset.lat[::10], lon=dataset.lon[::10])
+    result = dataset.interp(lat=dataset.lat[::10], lon=dataset.lon[::10])
```

Comment:
This results in discontinuous bounds. How about just using the first 10 points or so?

It would also be good to consider the case of ocean variables, e.g. `tos`, which will typically have 2D lat/lon coordinates.

Reply:
Didn't get the chance to get around to this today

Reply:
I've done something super niave for now, but lets address as part of #9